src/System.Linq/src/System/Linq/AppendPrepend.cs

```diff
@@ -352,12 +378,36 @@ public override AppendPrependIterator<TSource> Prepend(TSource item)
                 return new AppendPrependN<TSource>(_source, _prepended != null ? _prepended.Add(item) : new SingleLinkedNode<TSource>(item), _appended);
             }
 
+            private TSource[] LazyToArray()
+            {
+                Debug.Assert(GetCount(onlyIfCheap: true) == -1);
+
+                var builder = new LargeArrayBuilder<TSource>(initialize: true);
+
+                for (SingleLinkedNode<TSource> node = _prepended; node != null; node = node.Linked)
+                {
+                    builder.Add(node.Item);
+                }
+
+                builder.AddRange(_source);
+
+                if (_appended != null)
+                {
+                    foreach (TSource item in _appended.ToArray())
```

Comment:
@stephentoub [Yes.](https://github.com/jamesqo/corefx/blob/6c7675111fb3854b0e5535779a1dd04b8f4bd833/src/System.Linq/src/System/Linq/AppendPrepend.cs#L360) The logic in this method is basically a copy verbatim of what `MoveNext` currently does.